backend/app/routes.py

```diff
@@ -11,14 +11,32 @@
 
 @app.route('/')
 def index():
+    """Provide basic documentation about the available resources.
+    
+    All endpoints return JSON-formatted data.
+    """
     return jsonify({
-        'description': 'Travel Time App backend root',
-        'endpoints': [str(rule) for rule in app.url_map.iter_rules()]
+        'description': 'Travel Time App backend',
+        'available_endpoints': [
+            {
+                'path': str(rule),
+                'docstring': app.view_functions[rule.endpoint].__doc__
+            } for rule in app.url_map.iter_rules()
+        ]
     })
 
 # test URL /closest-node/-79.3400/43.6610
 @app.route('/nodes-within/<meters>/<longitude>/<latitude>', methods=['GET'])
-def closest_node(meters,longitude,latitude):
+def closest_node(meters, longitude, latitude):
+    """Return up to 20 nodes within a given radius (in meters) of a point.
+
+    Nodes are drawn from the Congestion Network, i.e. are fairly major intersections.
+
+    Arguments:
+    meters (numeric): distance around latitude and longitude to search
+    latitude (numeric): latitude of point to search around
+    longitude (numeric): longitude of point to search around

```

Comment:
Just a stylistic thing (and perhaps not too important). Numeric is not a type in python. I wonder if there is an actual name for a base class that encompasses both ints and floats. You could also write float, considering that you convert to float in your code anyway.

Reply:
Changed to float